ui/shared/DropdownMenuItems.jsx

```diff
@@ -4,13 +4,18 @@ import { DropdownMenu, DropdownItem } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faCheck } from '@fortawesome/free-solid-svg-icons';
 
-const DropdownMenuItems = ({ selectedItem, updateData, options }) => (
-  <DropdownMenu className="overflow-auto dropdown-menu-height">
+const DropdownMenuItems = ({
+  selectedItem,
+  updateData,
+  options,
+  dropdownNamespace = '',
+}) => (
+  <DropdownMenu className="overflow-hidden dropdown-menu-height">
     {options.map(item => (
       <DropdownItem
         tag="a"
-        key={item}
-        onClick={event => updateData(event.target.innerText)}
+        key={`${dropdownNamespace}${item}`}
+        onClick={event => updateData(event.target.textContent)}
```

Comment:
Each dropdown item knows exactly its inner text content, via the `item` variable.
We don't need to inspect the `event`'s content.

Reply:
During unit testing, when clicking an item from dropdown, the `event.target.innerText` is `undefined`, because `jest` does not know about events' `innerText`. More details [here](https://stackoverflow.com/questions/47902335/innertext-is-undefined-in-jest-test). In real world the code which filters alerts based on framework receives the name of the framework, but unit testing code receives `undefined`. So I used `textContent`. More on `textContent` [here](https://stackoverflow.com/questions/25429521/innertext-vs-textcontent-getting-some-errors-on-firefox).

Reply:
I agree with Ionut here. this is also the root issue for the `undefined` check that you had to do. If you click on a new option from the dropdown (anything else other that the selected element), it works well with `innerText` and `innerContent`, but if you click on the already selected option, it is confused by the `check` (:heavy_check_mark:)  symbol. This all thing  can be solved with:
`onClick={event => updateData(item)}`
as Ionut pointed out above

Reply:
You're right. I will remove the `undefined` check and use `item`.

Reply:
Using `item` instead of `event...` gives me a `Something went wrong Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.`. WIP here, I'm trying a way to solve this.

Reply:
Solved it.